utils/parsing.py

```diff
@@ -313,3 +315,117 @@ def brat_parse_to_bigbio_kb(brat_parse: Dict, entity_types: Iterable[str]) -> Di
             unified_example["coreferences"].append({"id": id_prefix + str(i), "entity_ids": entity_ids})
 
     return unified_example
+
+
+def parse_pubtator_file(text_or_filepath: str, is_filepath: bool = True) -> Iterator[Dict]:
+    """Returns a generator that returns a dict for each entry in `text_or_filepath`. If
+    `is_filepath`, we assume this is a filepath to a text file in PubTator format. Otherwise,
+    We assume this is a string containing the text in PubTator format.
+    """
+    if is_filepath:
+        f = open(text_or_filepath, "r")
+    else:
+        f = io.StringIO(text_or_filepath)
+
+    line = f.readline().strip()
+    while line != "":
```

Comment:
there might be a few other things going on, but it looks like the first lines of the `train` and `dev` files are blank and so nothing inside this `while` clause is executed.